app/coffee/Features/Compile/ClsiManager.coffee

```diff
@@ -101,56 +118,101 @@ module.exports = ClsiManager =
 		return outputFiles
 
 	VALID_COMPILERS: ["pdflatex", "latex", "xelatex", "lualatex"]
+
 	_buildRequest: (project_id, options={}, callback = (error, request) ->) ->
-		Project.findById project_id, {compiler: 1, rootDoc_id: 1, imageName: 1}, (error, project) ->
+		Project.findById project_id, {}, (error, project) ->
 			return callback(error) if error?
 			return callback(new Errors.NotFoundError("project does not exist: #{project_id}")) if !project?
-
+			console.log "PROJECT", project, JSON.stringify(project.rootFolder,null,2)
 			if project.compiler not in ClsiManager.VALID_COMPILERS
 				project.compiler = "pdflatex"
 
+			ClsiStateManager.checkState project_id, project, (error, stateOk, state) ->
+				return callback(error) if error?
+				logger.log project_id: project_id, checkState: stateOk, "checked project state"
+				console.log "OPTIONS ARE", options
+				if stateOk and not options.state? # incremental
+					ClsiManager._getContentFromDocUpdater project_id, (error, docUpdaterDocs) ->
+						return callback(error) if error?
+						# make this incremental
+						ProjectEntityHandler.getAllDocPathsFromProject project, (error, docPath) ->
```

Comment:
For an incremental compile we shouldn't need the files?  The files on disk in the clsi won't be removed.